### PR TITLE
SnapSDK Login

### PR DESCRIPTION
**Pull request by davidozhang**

### CODE REVIEW

File: SnapWar.xcodeproj/xcuserdata/pialeung.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 1]
```diff
@@ -12,7 +12,7 @@
 		<key>SnapWarShareExtension.xcscheme_^#shared#^_</key>
 		<dict>
 			<key>orderHint</key>
-			<integer>1</integer>
```
Comment (davidozhang) [inline SnapWar.xcodeproj/xcuserdata/pialeung.xcuserdatad/xcschemes/xcschememanagement.plist:15]: can you remove this file from version control? `git rm --cached path/to/file` and then add `SnapWar.xcodeproj/xcuserdata` to `.gitignore`

File: SnapWar/ViewControllers/SWAuthViewController.swift [hunk 2]
```diff
@@ -25,5 +26,42 @@ class SWAuthViewController: UIViewController, UITextFieldDelegate {
...
+        
+        SCSDKLoginClient.fetchUserData(
+                withQuery: graphQLQuery,
+                variables: nil,
+                success: { userInfo in
+                    if let userInfo = userInfo,
+                       let jsonData = try? JSONSerialization.data(withJSONObject: userInfo, options: .prettyPrinted) {
+                        if let data = String(data: jsonData, encoding: .ascii) {
+                            print("Fetched user info: " + data)
+                            // TODO: Assign data to user model
```
Comment (misspia) [inline SnapWar/ViewControllers/SWAuthViewController.swift:60]: nice, I can pick this up :D

File: SnapWar/Auth/SWLoginStatusObserver.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import SCSDKLoginKit
+
+class SWLoginStatusObserver: NSObject, SCSDKLoginStatusObserver {
```
Comment (misspia) [inline SnapWar/Auth/SWLoginStatusObserver.swift:3]: Where did you find the instructions to add this in?

**Review by misspia — APPROVED** [review]
Thankss 😺

File: SnapWar.xcodeproj/xcuserdata/pialeung.xcuserdatad/xcschemes/xcschememanagement.plist (hunk 1 above)
Comment (misspia) [reply SnapWar.xcodeproj/xcuserdata/pialeung.xcuserdatad/xcschemes/xcschememanagement.plist:15]: done 🙆‍♀️

File: SnapWar/ViewControllers/SWAuthViewController.swift (hunk 2 above)
Comment (davidozhang) [reply SnapWar/ViewControllers/SWAuthViewController.swift:60]: 😸

File: SnapWar/Auth/SWLoginStatusObserver.swift (hunk 3 above)
Comment (davidozhang) [reply SnapWar/Auth/SWLoginStatusObserver.swift:3]: no explicit instructions, i just looked up the interface of the observer and read the comments.. the documentation is like non-existent

Comment (davidozhang) [reply SnapWar/Auth/SWLoginStatusObserver.swift:3]: @misspia you can use this to watch for login status changes, so you can actually register stuff with this observer if you need

Comment (davidozhang) [reply SnapWar/Auth/SWLoginStatusObserver.swift:3]: things that might be useful are to know when to show the bitmoji avatars and maybe logout buttons

Comment (misspia) [reply SnapWar/Auth/SWLoginStatusObserver.swift:3]: okay got it :D

**Pull request merged**